codeclimate-wrapper

```diff
@@ -18,32 +18,52 @@ EOF
   exit 1
 }
 
+docker_for_mac() {
+  test -S /var/tmp/docker.sock > /dev/null 2>&1
+}
+
+socket_exists() {
+  test -S $socket_path > /dev/null 2>&1
+}
+
+socket_path() {
```

Comment:
📝  The current version of Docker for Mac (v1.11.1-beta10) symlinks `/var/run/docker.sock` to `/var/tmp/docker.sock`, so the changes which are really needed in this script is:
* setting `$socket_path` to the right path; and
* shortcircuiting for Docker for Mac users before testing for `boot2docker`.

The test for `docker-machine` are guarded by `if [ -n "$DOCKER_MACHINE_NAME" ]` and the [Docker for Mac documentation](https://beta.docker.com/docs/mac/docker-toolbox/#setting-up-to-run-docker-for-mac) directs the user to unset every environment variable prefixed with `DOCKER_`.